src/service/viewport/viewport-impl.js

```diff
@@ -554,7 +557,10 @@ export class ViewportImpl {
 
   /** @override */
   getScrollingElement() {
-    return this.binding_.getScrollingElement();
+    if (this.scrollingElement_) {
+      return this.scrollingElement_;
+    }
+    return (this.scrollingElement_ = this.binding_.getScrollingElement());
```

Comment:
Minor: Better to cache inside each binding implementation. Readers need to inspect `getScrollingElement()` to see if this is safe.